src/components/schedule/schedule-time-discover/ScheduleTimeSlot.tsx

```diff
@@ -57,6 +66,23 @@ const ScheduleTimeSlot: FC<ScheduleTimeSlotProps> = ({
     : false
 
   const { isTopElement, isBottomElement } = getMeetingBoundaries(slot, duration)
+
+  // Get current user event and URL from slotData
+  const currentUserEvent = slotData.currentUserEvent
+  const eventUrl = slotData.eventUrl
+
+  // Create a mapping from displayName to account address to identify current user
+  const displayNameToAddress = useMemo(() => {
```

Comment:
Isn't this the same for all slots? If yes, why isn't it in the parent scope? I can't imagine we'll need to calculate this for each slot